### PR TITLE
ci: pull cached images from local registry

### PR DESCRIPTION
**Pull request by nixpanic**
This reduces the number of images that get pulled from Docker Hub where
pull limits affect the jobs.

The `ceph/ceph:v15` image has been manually pushed to the CI repository. This needs further automation at a later time.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

### CODE REVIEW

File: mini-e2e.groovy [hunk 1]
```diff
@@ -94,11 +95,23 @@ node('cico-workspace') {
...
+			}
+
+			// base_image is like ceph/ceph:v15
+			ssh "podman pull ${ci_registry}/${base_image} && podman tag ${ci_registry}/${base_image} ${base_image}"
+			// cephcsi:devel is used with 'make containerized-build'
+			ssh "podman pull ${ci_registry}/cephcsi:devel"
+		}
 		stage('build artifacts') {
 			// build container image
 			ssh 'cd /opt/build/go/src/github.com/ceph/ceph-csi && make image-cephcsi GOARCH=amd64 CONTAINER_CMD=podman'
```
Comment (Madhu-1) [inline mini-e2e.groovy:112]: This will also try to pull the image. can we set the `BASE_IMAGE` in build.env so that it will be used as a base image?

Comment (nixpanic) [reply mini-e2e.groovy:112]: the image is already pulled, so it should not pull it again

File: mini-e2e.groovy [hunk 2]
```diff
@@ -94,11 +95,23 @@ node('cico-workspace') {
...
 		}
+		stage('pull base container images') {
+			def base_image = ssh 'source /opt/build/go/src/github.com/ceph/ceph-csi/build.env && echo ${BASE_IMAGE}'
+
+			withCredentials([usernamePassword(credentialsId: 'container-registry-auth', usernameVariable: 'CREDS_USER', passwordVariable: 'CREDS_PASSWD')]) {
+				ssh "podman login --user=${CREDS_USER} --password='${CREDS_PASSWD}'"
+			}
+
+			// base_image is like ceph/ceph:v15
+			ssh "podman pull ${ci_registry}/${base_image} && podman tag ${ci_registry}/${base_image} ${base_image}"
```
Comment (Madhu-1) [inline mini-e2e.groovy:106]: we are tagging it here to change the base image URL

**Review by Madhu-1 — APPROVED** [review]

File: mini-e2e.groovy (hunk 2 above)
Comment (nixpanic) [reply mini-e2e.groovy:106]: yes, this prevents pulling the base image during `make`, as it is already available on the local machine

**Pull request merged**